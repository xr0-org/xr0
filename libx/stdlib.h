#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
