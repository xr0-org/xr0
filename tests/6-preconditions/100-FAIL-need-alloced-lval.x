#include <stdlib.h>

int
func(int *x) ~ [ 
	pre: .alloc x;
	*x = 5;
]{
	*x = 5;	
}

int
main()
{
	int p;
	func(&p);	/* ERROR: spec requires heap */
}
