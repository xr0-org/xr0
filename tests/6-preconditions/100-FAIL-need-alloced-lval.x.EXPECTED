6-preconditions/100-FAIL-need-alloced-lval.x:15:10: cannot exec statement: parameter 'x' must be heap allocated
